Requests Tutorial

In [2]:
pip install requests

In [3]:
import requests

url = "https://www.example.com"
response = requests.get(url)

In [4]:
# it will show the HTTP status code
print(response)

<Response [200]>


In [5]:
print(response.status_code)

200


In [6]:
import requests

response = requests.get("https://www.example.com")
print(repsonse.content)

NameError: name 'repsonse' is not defined

Tu as une erreur dans ton script Python : tu as écrit repsonse au lieu de response

In [7]:
import requests

response = requests.get("https://www.example.com")
print(response.content)


b'<!doctype html>\n<html>\n<head>\n    <title>Example Domain</title>\n\n    <meta charset="utf-8" />\n    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />\n    <meta name="viewport" content="width=device-width, initial-scale=1" />\n    <style type="text/css">\n    body {\n        background-color: #f0f0f2;\n        margin: 0;\n        padding: 0;\n        font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;\n        \n    }\n    div {\n        width: 600px;\n        margin: 5em auto;\n        padding: 2em;\n        background-color: #fdfdff;\n        border-radius: 0.5em;\n        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);\n    }\n    a:link, a:visited {\n        color: #38488f;\n        text-decoration: none;\n    }\n    @media (max-width: 700px) {\n        div {\n            margin: 0 auto;\n            width: auto;\n        }\n    }\n    </style>    \n</head>\n\n<body>\n<div>\n    

In [9]:
data = {"name": "Salah", "message": "Hello!"}
url = "https://httpbin.org/post"

response = requests.post(url, json=data)

In [10]:
response_data = response.json()
# Shows the data as a dictionary
print(response_data)

{'args': {}, 'data': '{"name": "Salah", "message": "Hello!"}', 'files': {}, 'form': {}, 'headers': {'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate, zstd', 'Content-Length': '38', 'Content-Type': 'application/json', 'Host': 'httpbin.org', 'User-Agent': 'python-requests/2.32.3', 'X-Amzn-Trace-Id': 'Root=1-680a0b53-703fd6cd483293fa7b6ca1af'}, 'json': {'message': 'Hello!', 'name': 'Salah'}, 'origin': '34.73.195.247', 'url': 'https://httpbin.org/post'}


In [11]:
import requests

# here we use an endpoint that always gives a 404 status error
response = requests.get("https://httpbin.org/status/404")
# if status code is not 200 (successful response), then show error message
if response.status_code != 200:
    print(f"HTTP Error: {response.status_code}")

HTTP Error: 404


In [12]:
url = "https://httpbin.org/delay/10"

try:
    response = requests.get(url, timeout=5)
except requests.exceptions.Timeout as err:
    print(err)

HTTPSConnectionPool(host='httpbin.org', port=443): Read timed out. (read timeout=5)


In [13]:
auth_token = "XXXXXXXX"

# here we set the authorization header with the 'bearer token' for authentication purposes.
headers = {
    "Authorization": f"Bearer {auth_token}"
}

url = "https://httpbin.org/headers"
response = requests.get(url, headers=headers)
print(response.json())

{'headers': {'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate, zstd', 'Authorization': 'Bearer XXXXXXXX', 'Host': 'httpbin.org', 'User-Agent': 'python-requests/2.32.3', 'X-Amzn-Trace-Id': 'Root=1-680a0b9c-246e2ab051f94ee67e2438f9'}}


In [14]:
import requests

url = "https://www.example.com"
# this will get all the HTML, javascript, css code
response = requests.get(url)

In [15]:
pip install beautifulsoup4

In [17]:
import requests
from bs4 import BeautifulSoup

url = "https://www.example.com"
response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")

In [18]:
title = soup.title.text
content = soup.find("p").text
links = [a["href"] for a in soup.find_all("a")]

print(title, content, links)

Example Domain This domain is for use in illustrative examples in documents. You may use this
    domain in literature without prior coordination or asking for permission. ['https://www.iana.org/domains/example']


In [19]:
import urllib.request
import urllib.parse

data = urllib.parse.urlencode({"key"; "value"}).encode("utf-8")
req = urllib.request.Request("https://www.example.com", data=data, method="post")
with urllib.request.urlopen(req) as response:
    html = response.read().decode("utf-8")
print(html)

SyntaxError: invalid syntax (<ipython-input-19-5ca3ef0be473>, line 4)

L'erreur vient de l'utilisation incorrecte du point-virgule (;) au lieu du deux-points (:) dans ton dictionnaire Python.

In [20]:
import urllib.request
import urllib.parse

data = urllib.parse.urlencode({"key": "value"}).encode("utf-8")  # <-- correction ici
req = urllib.request.Request("https://www.example.com", data=data, method="POST")
with urllib.request.urlopen(req) as response:
    html = response.read().decode("utf-8")
print(html)


HTTPError: HTTP Error 403: Forbidden

Le message d’erreur HTTP Error 403: Forbidden signifie que le serveur (ici https://www.example.com) refuse la requête

In [21]:
import urllib.request
import urllib.parse

url = "https://httpbin.org/post"  # URL de test qui accepte les requêtes POST
headers = {
    "User-Agent": "Mozilla/5.0",  # Simule un navigateur
    "Content-Type": "application/x-www-form-urlencoded"
}

data = urllib.parse.urlencode({"key": "value"}).encode("utf-8")
req = urllib.request.Request(url, data=data, headers=headers, method="POST")

with urllib.request.urlopen(req) as response:
    html = response.read().decode("utf-8")

print(html)


{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {
    "key": "value"
  }, 
  "headers": {
    "Accept-Encoding": "identity", 
    "Content-Length": "9", 
    "Content-Type": "application/x-www-form-urlencoded", 
    "Host": "httpbin.org", 
    "User-Agent": "Mozilla/5.0", 
    "X-Amzn-Trace-Id": "Root=1-680a0c76-559e3cc777c02c004df34f5a"
  }, 
  "json": null, 
  "origin": "34.73.195.247", 
  "url": "https://httpbin.org/post"
}

